pytest/tests/sanity/memtrie_disktrie_switch.py

```diff
@@ -0,0 +1,342 @@
+        """Returns a next target height until which we will send the transactions."""
+        current_height = self.__wait_for_blocks(1)
+        stop_height = random.randint(1, EPOCH_LENGTH)
+        return current_height + num_epochs * EPOCH_LENGTH + stop_height
+
+    def next_nonce(self, signer_key):
+        """Returns the next nonce to use for sending transactions for the given signing key."""
+        assert signer_key in self.nonces
+        nonce = self.nonces[signer_key]
+        self.nonces[signer_key] = nonce + 42
+        return nonce
+
+    def __restart_nodes(self, enable_memtries):
+        """Stops and restarts the nodes with the config that enables/disables memtries.
+        
+        It restarts only the validator nodes and does NOT restart the RPC node."""
+        boot_node = self.rpc_node
+        for i in range(0, 4):
+            self.nodes[i].kill()
+            time.sleep(2)
+            self.nodes[i].change_config(
+                {"store.load_mem_tries_for_tracked_shards": enable_memtries})
+            self.nodes[i].start(boot_node=None if i == 0 else boot_node)
+
+    def __random_workload_until(self, target_height):
+        """Generates traffic to make transfers between accounts."""
+        last_height = -1
+        while True:
+            last_block = self.rpc_node.get_latest_block()
+            height = last_block.height
+            if height > target_height:
+                break
+            if height != last_height:
+                logger.info(
+                    f'@{height}, epoch_height: {state_sync_lib.approximate_epoch_height(height, EPOCH_LENGTH)}'
+                )
+                last_height = height
+            last_block_hash = last_block.hash_bytes
+            if random.random() < 0.5:
+                # Make a transfer between accounts.
+                # The goal is to generate cross-shard receipts.
+                from_account_key = random.choice(self.account_keys)
+                to_account_id = random.choice([
+                    account_key.account_id
+                    for account_key in self.account_keys
+                    if account_key.account_id != from_account_key.account_id
+                ] + ["near"])
+                payment_tx = transaction.sign_payment_tx(
+                    from_account_key, to_account_id, 1,
+                    self.next_nonce(from_account_key), last_block_hash)
+                result = self.rpc_node.send_tx(payment_tx)
+                assert 'result' in result and 'error' not in result, (
+                    'Expected "result" and no "error" in response, got: {}'.
+                    format(result))
+                logger.debug("Transfer: {}".format(result))
+                tx_hash = result['result']
+                self.txs.append((from_account_key.account_id, tx_hash))
+            elif len(self.keys) > 10 and random.random() < 0.5:
+                # Do some flat storage reads, but only if we have enough keys populated.
```

Comment:
nit: I suggest replacing flat storage with just storage here and at line 205: for disktrie flat storage is just an implementation detail and for memtrie it is not even relevant